### DISCUSSION TITLE
Should ClickHouse be user facing for high frequency of requests?

### DISCUSSION THREAD

**Discussion by Drabuna** [ClickHouse/ClickHouse#36688] [Q&A] [answered]
Hi,

I'm building an event collection and analytics system on top of ClickHouse.
There's going to be a lot of data, and a certain number of analytical "reports" that users can access. 

So far the setup for each of such reports is as follows:
1. A `MergeTree` table with TTL for collecting raw incoming events.
2. `MaterializedView` that transforms/aggregates data and puts it to `AggregatingMergeTree` (long-term store for transformed data)
3. `View` that wraps data from the `AggregatingMergeTree`.

Now, the users will be hitting that `View` and similar other views when they need to fetch data for their dashboards.
The average query latency on those is around 50-60 ms. 
As far as I understood from documentation a lot of small queries are not recommended.

The question is - before going live with this, should another layer be built in front of ClickHouse, something like a MySQL cache for example, that will query ClickHouse -> store dashboard data in MySQL -> return it to the user. This cache can be invalidated once a day, and fresh data can be fetched from ClickHouse. 
Can I somehow utilize ClickHouse <-> MySQL integration for this?

Or should I skip this caching layer altogether?

Thanks!

**Comment by alexey-milovidov** [comment, marked as answer]
ClickHouse is good to serve direct queries from users.

One example is Metrica: https://w3techs.com/technologies/overview/traffic_analysis, it collects around 5% of all internet traffic (more than 100 billion events every day) and serving realtime reports for customers directly from ClickHouse.

Although the number of SELECT queries is not very high, less than 50 per second per server.
The latency of SELECT queries are from 5ms to a few seconds, typically below 100ms.

Quotas are used to limit the amount of SELECT queries per user.

**Reply by Drabuna** [reply]
Ah, cool! Thank you! :)
